runtime/src/tiered_storage.rs

```diff
@@ -43,6 +45,7 @@ pub struct TieredStorageFormat {
 
 #[derive(Debug)]
 pub struct TieredStorage {
+    reader: OnceCell<TieredStorageReader>,
```

Comment:
Since the current design is Writer, Reader, and TieredStorage, it probably makes more sense to the current design to get more things reviewed, and along the way, we can find ways to remove the `TieredStorage` struct, plus TieredStorage doesn't have too many functions.  The main challenge should be the limitation cast by the `AccountStorageEntry`.

Reply:
The thing is that the struct actually doesn't keep the Writer instance.  It only has Option<Format> and OnceCell<Reader>, and we want to clear all the resources occupied by the writer once the `write_accounts()` call completes.